### DISCUSSION TITLE
Next possible steps for the MetaAya project...

### DISCUSSION THREAD

**Discussion by lunalunaa** [aya-prover/aya-dev#375] [Ideas]
Priority 0 (enhancements):
- [x] More tests
- [x] Separate everything related to tactic out into another class, something like `TacticElaborator`
- [ ] LSP integration

-------

Priority 1 (new features):
- [ ] Implement syntactic reflection and quasiquoting
- [ ] Idris-style monadic elaborator reflection (requires a monadic infrastructure)
- [ ] Coq/lean-style tactic

--------

Priority 2 (research):
- [ ] Performance benchmarking and optimization
- [ ] MTac(2)-style well-typed metaprogramming

-------

Priority ∞:
- [ ] Formalized metatheory of Aya, in Aya: Aya Aya Accept!

**Comment by ice1000** [comment]
I propose `TacticElaborator` -> `TacticTycker` or `TacTycker`
